Object Store Doctest Failure with Default Features

**Describe the bug**
<!--
A clear and concise description of what the bug is.
-->

Running `cargo test` on the object_store crate result in a failing doctest

```
---- src/lib.rs - (line 105) stdout ----
error[E0432]: unresolved import `object_store::aws`
 --> src/lib.rs:109:19
  |
7 | use object_store::aws::{AmazonS3, AmazonS3Builder};
  |                   ^^^ could not find `aws` in `object_store`
```

This was introduced in #4978 and not caught by CI as we only test with all features

**To Reproduce**
<!--
Steps to reproduce the behavior:
-->

**Expected behavior**
<!--
A clear and concise description of what you expected to happen.
-->

**Additional context**
<!--
Add any other context about the problem here.
-->

https://github.com/apache/arrow-rs/pull/5024 worked around this for the purposes of release verification, and is also generally a more correct for the verification scripts to be doing.

--- a/object_store/Cargo.toml
+++ b/object_store/Cargo.toml
@@ -67,10 +67,10 @@ http = ["cloud"]
 tls-webpki-roots = ["reqwest?/rustls-tls-webpki-roots"]
 
 [dev-dependencies] # In alphabetical order
-tempfile = "3.1.0"
 futures-test = "0.3"
-rand = "0.8"
 hyper = { version = "0.14.24", features = ["server"] }
+rand = "0.8"
+tempfile = "3.1.0"
 
 [[test]]
 name = "get_range_file"

--- a/object_store/src/lib.rs
+++ b/object_store/src/lib.rs
@@ -119,6 +119,7 @@
 //! application complexity.
 //!
 //! ```no_run
+//! # #[cfg(feature = "aws")] {
 //! # use url::Url;
 //! # use object_store::{parse_url, parse_url_opts};
 //! # use object_store::aws::{AmazonS3, AmazonS3Builder};
@@ -140,6 +141,7 @@
 //! let url = Url::parse("https://ACCOUNT_ID.r2.cloudflarestorage.com/bucket/path").unwrap();
 //! let (store, path) = parse_url(&url).unwrap();
 //! assert_eq!(path.as_ref(), "path");
+//! # }
 //! ```
 //!
 //! [PyArrow FileSystem]: https://arrow.apache.org/docs/python/generated/pyarrow.fs.FileSystem.html#pyarrow.fs.FileSystem.from_uri

